lib/scala/runtime-version-manager/src/main/scala/org/enso/runtimeversionmanager/runner/Command.scala

```diff
@@ -18,13 +20,25 @@ case class Command(command: Seq[String], extraEnv: Seq[(String, String)]) {
     * May return an exception if it is impossible to run the command (for
     * example due to insufficient permissions or nonexistent executable).
     */
-  def run(): Try[Int] =
+  def run(inheritStdOutErr: Boolean = true): Try[(Int, String)] =
     wrapError {
       logger.debug("Executing {}", this)
       val processBuilder = builder()
-      processBuilder.inheritIO()
-      val process = processBuilder.start()
-      process.waitFor()
+      if (inheritStdOutErr) {
+        processBuilder.inheritIO()
+        val process = processBuilder.start()
+        (process.waitFor(), "")
+      } else {
+        val process = processBuilder.start()
+        val outReader =
+          new BufferedReader(new InputStreamReader(process.getInputStream))
+        val errReader =
+          new BufferedReader(new InputStreamReader(process.getErrorStream))
+        val joiner = new StringJoiner(System.getProperty("line.separator"))
+        outReader.lines.iterator.forEachRemaining(joiner.add)
+        errReader.lines.iterator.forEachRemaining(joiner.add)
+        (process.waitFor(), joiner.toString)
+      }
     }
```

Comment:
I'm not a fan of this API. In most invocations we discard the returned `""`...

Could we please keep `run(): Try[Int]` as is and just add a separate one `runCapturingOutput(): Try[(Int, String)]` ?

Reply:
Also keep in mind that this will likely get a race condition if the amount of output is larger - we are reading the streams _after_ `waitFor` exits. But if the internal system buffers get full, the child process trying to write to the streams will block waiting for us to read something - and we will keep waiting for child process to exit - deadlock.

We have to keep reading the buffers in some background thread to avoid this lock scenario.
One implementation of that is in `lib/scala/testkit/src/main/scala/org/enso/testkit/process/WrappedProcess.scala`, you could move it to a non-test library and use it here.

I see even our `System::createProcess` implementation is only partially correct - it reads the streams before `waitFor` but it does so sequentially - but the only correct solution is to read them concurrently, either through Threads or `select`-like multiplexing - as it is possible that e.g. the stderr stream will get saturated while the stdout is not closed yet.

Reply:
Sure I can change the API.

Re WrappedProcess: I can see that you've added quite an elaborate logic to handle streams. Sure, I can use that. Thanks for the pointer.